vertx-mutiny-clients/vertx-mutiny-sql-client/src/main/java/io/vertx/mutiny/sqlclient/SqlClientHelper.java

```diff
@@ -0,0 +1,100 @@
+package io.vertx.mutiny.sqlclient;
+
+import java.util.function.Function;
+
+import io.smallrye.mutiny.Multi;
+import io.smallrye.mutiny.Uni;
+
+/**
+ * Utilities for generating {@link Multi} and {@link Uni} with a {@link SqlClient}.
+ */
+public class SqlClientHelper {
+
+    /**
+     * Generates a {@link Multi} from operations executed inside a {@link Transaction}.
+     *
+     * @param pool the {@link Pool}
+     * @param sourceSupplier a user-provided function returning a {@link Multi} generated by interacting with the given
+     *        {@link Transaction}
+     * @param <T> the type of the items emitted by the {@link Multi}
+     * @return a {@link Multi} generated from operations executed inside a {@link Transaction}
+     */
+    public static <T> Multi<T> inTransactionMulti(Pool pool, Function<Transaction, Multi<T>> sourceSupplier) {
+        return usingConnectionMulti(pool, conn -> {
+            Transaction transaction = conn.begin();
+            return Multi.createBy().concatenating().streams(
+                    sourceSupplier.apply(transaction),
+                    // TODO: find a better way to transform a Uni<Void> to an empty Multi<T>
+                    transaction.commit().toMulti().transform().byTakingFirstItems(0).onItem().apply(aVoid -> (T) aVoid)).on()
```

Comment:
What about `Uni.createFrom().nullValue().toMulti()` ?

Reply:
Or `transaction.commit().map(v -> (T) v).toMulti()` maybe?

Reply:
Not sure it's useful here, but there is an `onItemOrFailure` group that gives you both events (well either one or the other) but it let you handle them in on _go_.

Reply:
I went with `transaction.commit().onItem().produceMulti(v -> Multi.createFrom().nothing())`

Reply:
Interesting, tests don't pass with this one